lib/run/run.sh

```diff
@@ -144,20 +160,26 @@ bpkg_run () {
       # shellcheck disable=SC1090
       source "$(which "$name")"
     else
-      if test -f bpkg.json || test -f package.json; then
-        BPKG_SCRIPT_SOURCES=$(find . -name '*.sh')
-        export BPKG_SCRIPT_SOURCES
-      fi
-
-      export BPKG_NAME="$(bpkg_package name)"
-      export BPKG_REPO="$(bpkg_package repo)"
-      export BPKG_VERSION="$(bpkg_package version)"
+      # shellcheck disable=SC2230
+      # shellcheck source=lib/env/env.sh
+      source "$(which bpkg-env)"
 
       if [ -n "$cmd" ]; then
+        local parts=()
+        # shellcheck disable=SC2086
+        read -r -a parts <<< $cmd
+        local args=()
+        local prefix="${parts[0]}"
+
+        for (( i = 1; i < ${#parts[@]}; i++ )); do
+          if [[ "${parts[$i]}" =~ \*.\* ]]; then
+            args+=($(find . -wholename "${parts[$i]}"))
```

Comment:
Instead of `-wholename` you might want to use `-path` as it does the same, but is more portable (`-wholename` is GNU-specific, [`-path` is POSIX](https://www.gnu.org/software/findutils/manual/html_node/find_html/Full-Name-Patterns.html))